lemminx-maven/src/main/java/org/eclipse/lemminx/maven/PluginValidator.java

```diff
@@ -54,28 +74,38 @@ public Diagnostic validateGoal(DiagnosticRequest diagnosticRequest) {
 			return null;
 		}
 		if (node.isElement() && node.hasChildNodes()) {
-			List<Parameter> parameters = MavenPluginUtils.collectPluginConfigurationParameters(diagnosticRequest, cache,
-					repoSession, pluginManager);
-			Diagnostic diag = internalValidateGoal(diagnosticRequest, parameters);
-			if (diag != null) {
-				diagnosticRequest.getDiagnostics().add(diag);
+			PluginDescriptor pluginDescriptor;
+			try {
+				pluginDescriptor = MavenPluginUtils.getContainingPluginDescriptor(diagnosticRequest, cache, repoSession,
+						pluginManager);
+				Diagnostic diag = internalValidateGoal(diagnosticRequest, pluginDescriptor);
+				if (diag != null) {
+					diagnosticRequest.getDiagnostics().add(diag);
+				}
+			} catch (PluginResolutionException | PluginDescriptorParsingException
+					| InvalidPluginDescriptorException e) {
+				e.printStackTrace();
```

Comment:
@mickaelistria do you think goals should also give a diagnostic if the plugin wasn't resolvable? Something similar to [line 51](https://github.com/eclipse/lemminx-maven/pull/54/files#diff-75b35d7fdfb0eab29a93b39bd6847f17R51)

Reply:
I think if we cannot get plugin info, we should only show one diagnostic for goal/configuration.

Reply:
So rather than have a diagnostic show up under goal & configuration, only make the diagnostic show on one of them? 
Ideally, I should try and make the plugin's GAV tags get this "unresolvable plugin" warning diagnostic.

Reply:
I ended up making a plugin's <artifactId> tag get a diagnostic if a plugin could not be resolved. In this case, all configuration/goal diagnostics are disabled.